actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -53,6 +58,9 @@ class ActionCable.Connection
     @isState("open", "connecting")
 
   # Private
+  isProtocolSupported: ->
+    [..., unsupported_protocol] = protocols
+    @getProtocol() in protocols and @getProtocol() isnt unsupported_protocol
```

Comment:
I had a back and forth with @kaspth about this... he wanted to minimize the heavy separation I made in the code between the supported and unsupported protocols. However, I don't think your suggestion conflicts with what he was looking for, so I made the change.